src/models/userMedia.ts

```diff
@@ -23,28 +27,124 @@ export enum Action {
 const isChrome: boolean = !!(navigator.userAgent.indexOf("Chrome") !== -1);
 console.log("IsChrome", isChrome);
 
+// NodeChain is responsible for creating, destroying, and connecting our
+// Web Audio API nodes.
+class NodeChain {
```

Comment:
This moves the audio graph creation and destruction into its own class from `UserMedia`, making it more readable and easier to cover in a straightforward way. It also handles proper disconnection of all the nodes.